etna/records/models.py

```diff
@@ -1,78 +1,340 @@
+
+    @cached_property
+    def catalogue_source(self) -> str:
+        return self.get("source.value", default="")
+
+    @property
+    def raw_description(self) -> str:
+        try:
+            return self.highlights["@template.details.description"]
+        except KeyError:
+            pass
+        try:
+            return self.template["description"]
+        except KeyError:
+            pass
+        description_items = self.get("description", ())
+        for item in description_items:
+            if item.get("type", "") == "description" or len(description_items) == 1:
+                return item.get("value", "")
+        return ""
+
+    @cached_property
+    def description(self) -> str:
+        if raw := self.raw_description:
+            return format_description_markup(raw)
+        return ""
+
+    @cached_property
+    def held_by(self) -> str:
+        return self.template.get("heldBy", "")
+
+    @cached_property
+    def origination_date(self) -> str:
+        return self.template.get("dateCreated", "")
+
+    @cached_property
+    def level(self) -> str:
+        return self.get("level.value", self.template.get("level", ""))
+
+    @cached_property
+    def level_code(self) -> int:
+        return self.get("level.code", None)
+
+    @cached_property
+    def availability_delivery_condition(self) -> str:
+        return self.template.get("deliveryOption", "")
 
     @property
     def availability_condition_category(self) -> str:
         return settings.AVAILABILITY_CONDITION_CATEGORIES.get(
             self.availability_delivery_condition, ""
         )
 
+    @cached_property
+    def repo_summary_title(self) -> str:
+        return self.get("repository.summary.title", default="")
+
+    @cached_property
+    def repo_archon_value(self) -> str:
+        for item in self.get("repository.identifier", ()):
+            if item["type"] == "Archon number":
+                return item.get("value", "")
+        return ""
+
+    @cached_property
+    def parent(self) -> Union["Record", None]:
+        if parent_data := self.get("parent.0", default=None):
+            return Record(parent_data)
+
+    @cached_property
+    def hierarchy(self) -> Tuple["Record"]:
+        return [
+            Record(item)
+            for item in self.get("@hierarchy.0", default=())
+            if item.get("identifier")
+        ]
+
+    @cached_property
+    def next_record(self) -> Union["Record", None]:
+        if next := self.get("@next", default=None):
+            return Record(next)
+
+    @cached_property
+    def previous_record(self) -> Union["Record", None]:
+        if prev := self.get("@previous", default=None):
+            return Record(prev)
+
+    @cached_property
+    def topics(self) -> Tuple[Dict[str, str]]:
+        return_value = []
+        for item in self.get("topic", default=()):
+            topic_title = ""
+            try:
+                topic_title = item.get["name"][0]["value"]
```

Comment:
Was not able to find the API response for topic, which API returns this path?

Reply:
The aim was to replicate what the transform function is doing. See original code here: https://github.com/nationalarchives/ds-wagtail/blob/develop/etna/records/transforms.py#L85-L94